app/services/filter_tagged_content.rb

```diff
@@ -0,0 +1,62 @@
+class FilterTaggedContent
```

Comment:
Could this class name be improved? It's currently phrased as a verb, which isn't very typical. I'm not 100% sure myself of what this class is intended to do (partly because of the naming), but given the amount of methods that start with `valid`, maybe something like `TaggedContentValidator` is more descriptive?